src/sentry/integrations/github/integration.py

```diff
@@ -155,7 +155,7 @@ def get_document_origin(org) -> str:
 # Github App docs and list of available endpoints
 # https://docs.github.com/en/rest/apps/installations
 # https://docs.github.com/en/rest/overview/endpoints-available-for-github-apps
-class GitHubIntegration(IntegrationInstallation, GitHubIssueBasic, RepositoryMixin, CommitContextMixin):  # type: ignore[misc]
+class GitHubIntegration(IntegrationInstallation, RepositoryIntegration, GitHubIssueBasic, CommitContextMixin):  # type: ignore[misc]
```

Comment:
i've only replaced the GitHub integration's `RepositoryMixin` with `RepositoryIntegration`, but i've added `RepositoryClient` as a parent class of all the SCM integration clients

Reply:
in this class i've also grouped the methods by the parent class methods they are overriding